samples/wifi/sta/src/main.c

```diff
@@ -317,3 +319,25 @@ int main(void)
 
 	return 0;
 }
+
+void wifi_ready_cb(void)
+{
+	LOG_INF("Wi-Fi is ready");
+	k_sem_give(&wifi_ready_sem);
+}
+
+int main(void)
+{
+	app_callback_t callback = {
+		.is_wifi_ready = wifi_ready_cb,
+	};
+
+	register_supp_events(callback);
+
+	LOG_INF("Waiting for Wi-Fi to be ready");
+	k_sem_take(&wifi_ready_sem, K_FOREVER);
```

Comment:
can we also add a demonstration for non-blocking call?

Reply:
As discussed offline, will handle this later.